### PR TITLE
UI 10/modal

### PR DESCRIPTION
**Pull request by maciek15**

### CODE REVIEW

File: packages/modal/src/Modal.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import PropTypes from 'prop-types'
+import cls from 'classnames'
+
+import { prefix } from '@talixo/commons'
+
+const name = prefix('modal')
```
Comment (rangoo94) [inline packages/modal/src/Modal.js]: `moduleName`? :)

File: packages/modal/src/Modal.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+const modalRoot = document.querySelector('body')
+
+/**
+ * Component which represents modal.
+ *
+ * @param {object} props
+ * @param {string} [props.className]
+ * @param {node} [props.children]
```
Comment (rangoo94) [inline packages/modal/src/Modal.js]: `*`, because `node` is not correct type in JS, it's only React propType

File: packages/modal/src/Modal.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+ * @param {boolean} [props.isOpen]
+ * @param {string} [props.size]
+ * @param {object} [props.style]
```
Comment (rangoo94) [inline packages/modal/src/Modal.js]: Does Portal have these properties?

File: packages/modal/src/Modal.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  toggleOpen () {
+    this.setState({ isOpen: !this.state.isOpen })
+  }
+
+  render () {
+    const { children, className, size, style, isOpen, ...rest } = this.props
+    return (
+      <div>
+        <ModalPortal>
+          <div className={cls(prefix('modal--backdrop'), { entered: this.state.isOpen, exiting: !this.state.isOpen })}>
```
Comment (rangoo94) [inline packages/modal/src/Modal.js]:
```
`${moduleName}-backdrop`
```

File: packages/modal/src/Modal.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+            <div
+              className={cls(name, className, {
+                [prefix(`modal--${size}`)]: size !== undefined
+              })}
+              style={style}
```
Comment (rangoo94) [inline packages/modal/src/Modal.js]: Is `style` required there? It's passed just like any other prop `{...rest}`

File: packages/modal/src/Modal.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+  className: PropTypes.string,
+
+  /** Content of modal */
+  children: PropTypes.node,
+
+  /** Controls whether modal is open */
+  isOpen: PropTypes.bool,
+
+  /** Size of the modal ('small', 'medium', 'large') */
+  size: PropTypes.string,
```
Comment (rangoo94) [inline packages/modal/src/Modal.js]: Not sure if we should have modal sizes

File: packages/modal/src/Modal.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+  /** Additional styles */
+  style: PropTypes.object
```
Comment (rangoo94) [inline packages/modal/src/Modal.js]: These styles are just passed down like any other prop

File: packages/modal/stories.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+const getInitialStateDefault = () => {
+  return {
+    isOpen: false
+  }
+}
+
+const getInitialStateSmall = () => {
+  return {
+    isOpen: false,
+    size: 'small'
```
Comment (rangoo94) [inline packages/modal/stories.js]: Actually `size` shouldn't be there, but it's just a story, so no problem :)

File: packages/modal/styles/geometry.sass [hunk 9]
```diff
@@ -0,0 +1,47 @@
+@import './config'
+
+@include module('modal')
+  width: 90%
+  max-width: 500px
+  padding: grid-unit(2)
+
+#{prefix('modal--small')}
```
Comment (rangoo94) [inline packages/modal/styles/geometry.sass]: or

```sass
@include module('modal')
  width: 90%
  max-width: 500px
  padding: grid-unit(2)

  &--small
    width: 60%
    max-width: 300px

  &--large
    width: 90%
    max-width: 500px
```

etc.

File: packages/modal/src/Modal.js (hunk 6 above)
Comment (maciek15) [reply packages/modal/src/Modal.js]: TBD

File: packages/modal/src/Modal.js [hunk 10]
```diff
@@ -0,0 +1,105 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import PropTypes from 'prop-types'
+import cls from 'classnames'
+
+import { prefix } from '@talixo/commons'
+
+const moduleName = prefix('modal')
+
+const modalRoot = document.querySelector('body')
```
Comment (rangoo94) [inline packages/modal/src/Modal.js]: Please do not use automatically stuff which will require browser (except stuff which will anyway require that)

File: packages/modal/src/Modal.js [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+ * @returns {React.Element}
+ */
+class ModalPortal extends React.Component {
+  constructor (props) {
+    super(props)
+    this.el = document.createElement('div')
+  }
+
+  componentDidMount () {
+    modalRoot.appendChild(this.el)
```
Comment (rangoo94) [inline packages/modal/src/Modal.js]: Could you pass `root` as prop?

File: packages/modal/README.md [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+
+## Supported props
+
+It allows any props which are allowed for `div`. Additionally, it handles some differently:
+
+Property name | Type      | Default | Description                    
+--------------|-----------|:-------:|--------------------------------
+className     | string    | n/a     | additional class name passed to wrapper
+children      | string    | n/a     | content of modal
+isOpen        | string    | `true`  | controls whether modal is open
```
Comment (PiotrGrzechnik) [inline packages/modal/README.md]: Shouldn't be type = bool? In Modal.propTypes isOpen is marked as boolean.

File: packages/modal/styles/theming.sass [hunk 13]
```diff
@@ -0,0 +1,16 @@
+@import './config'
+
+@include module('modal')
+  background-color: $color-white
+
+#{prefix('modal-backdrop')}
+  background-color: rgba(0, 0, 0, 0.3)
+  user-select: none
+
+
```
Comment (PiotrGrzechnik) [inline packages/modal/styles/theming.sass]: Unnecessary line :)

File: packages/modal/styles/geometry.sass [hunk 14]
```diff
@@ -0,0 +1,31 @@
+@import './config'
+
+@include module('modal')
+  width: 90%
+  max-width: 500px
```
Comment (rangoo94) [inline packages/modal/styles/geometry.sass]: `max-width` in `dp`

**Pull request merged**